(PORT)IPC SPECIES

ScarletLightweaverSS14 opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
A new unique species, IPC PORTED FROM FAR HORIZONS

## Why we need to add this
Roleplay Story telling player customization.
ORGINAL CREATORS / PR
https://github.com/Far-Horizons-SS14/Far-Horizons-SS14/pull/135
https://github.com/KillerTheFlareon
https://github.com/princess-gurchi

## Media (Video/Screenshots)
<img width="280" height="78" alt="image" src="https://github.com/user-attachments/assets/73fef5f6-49e1-4fa4-ac2d-d49e45f4581c" />

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [ ] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
NT has authorised the use of IPC species on stations

:cl: SCARLET LIGHTWEAVER, KillerTheFlareon, princess-gurchi
- add: NT has authorised the use of IPC species on stations.

ScarletLightweaverSS14 committed:
INITAL

ScarletLightweaverSS14 committed:
updates, fixes, features

ScarletLightweaverSS14 committed:
improvments and fixrs

CringeCursed reviewed (commented):
Why are you rebasing the Far Horizons code under the _Starlight folders?

CrazyPhantom779 commented:
What cringe said. Even more so with FH being a downstream of us, they will inevitably have very annoying merge conflicts so you should put them in a FH with the same name they use

ScarletLightweaverSS14 committed:
COMPLETE! i hope i may need to fix some things

ScarletLightweaverSS14 commented:
done(i think)

ScarletLightweaverSS14 committed:
Merge branch 'starlight-dev' into IPC

CorruptOmega commented:
Everything should remain under Far Horizons directories  except where we added to Starlight and base files.

You're just causing a lot of headaches downstream by moving everything to Starlight and wizden folders because then we'll get duplicates when we do upstream from SL

ScarletLightweaverSS14 commented:
> Everything should remain under Far Horizons directories except where we added to Starlight and base files.
> 
> You're just causing a lot of headaches downstream by moving everything to Starlight and wizden folders because then we'll get duplicates when we do upstream from SL

im working on moving all ported files to FH files

CorruptOmega commented:
> > Everything should remain under Far Horizons directories except where we added to Starlight and base files.
> > 
> > You're just causing a lot of headaches downstream by moving everything to Starlight and wizden folders because then we'll get duplicates when we do upstream from SL
> 
> im working on moving all ported files to FH files

Thank you very much. Sorry if it came off rough. Just had a mild panic attack thinking about sorting through that haha.

ScarletLightweaverSS14 committed:
THE FINALE

ScarletLightweaverSS14 committed:
fixes i hope thats all

ScarletLightweaverSS14 committed:
fixes

ScarletLightweaverSS14 committed:
patch

ScarletLightweaverSS14 committed:
patches

ScarletLightweaverSS14 committed:
build patches

ScarletLightweaverSS14 committed:
patch/final

ScarletLightweaverSS14 commented:
DONE!!!

wonderfulnewworld commented:
holy shit.... ipcs... just a day away...

DocMoth commented:
second time ive gotten to say ipcs tomorrow and mean it

CrazyPhantom779 commented:
Everything in upstream files needs comments

ScarletLightweaverSS14 commented:
> Everything in upstream files needs comments

consider it done sooon @CrazyPhantom779

ScarletLightweaverSS14 committed:
comments

ScarletLightweaverSS14 commented:
Fix Required
Adjust pricing in cargo prototypes to ensure sell price ≤ purchase cost. Either:

Increase purchase cost to ≥1438 credits, or
Decrease sell value multiplier/base price
Location
Check: Resources/Prototypes/Cargo/cargo_product.yml or similar pricing configuration for FoodHappyHonkMeals

@Happyrobot33  unrelated to PR

ScarletLightweaverSS14 committed:
Merge branch 'starlight-dev' into IPC

CringeCursed commented on Content.Server/Body/Systems/BodySystem.cs:
```diff
@@ -25,7 +25,7 @@ public sealed class BodySystem : SharedBodySystem
     [Dependency] private readonly MobStateSystem _mobState = default!;
     [Dependency] private readonly SharedMindSystem _mindSystem = default!;
-    [Dependency] private readonly LimbSystem _limbSystem = default!;//🌟Starlight🌟
+    [Dependency] private readonly LimbSystem _limbSystem = default!;
```
??????

CringeCursed commented on Content.Server/Database/ServerDbBase.cs:
```diff
@@ -333,7 +333,7 @@ private static HumanoidCharacterProfile ConvertProfiles(Profile profile)
                 profile.CharacterName,
                 profile.Voice,
-                profile.SiliconVoice, // 🌟Starlight🌟
+                profile.SiliconVoice,
```
??????

CringeCursed commented on Resources/Textures/Mobs/Species/IPC/ATTRIBUTION.txt:
![SB55oIPPAXc](https://github.com/user-attachments/assets/5b730226-ec3d-45f1-a1de-30a2dc860817)

t-t-t-t-txt atribution for TEXTURE FILE?
Metafiles are jokes now.

CringeCursed commented on Resources/Textures/Mobs/Customization/ipc/ATTRIBUTION.txt:
```diff
@@ -0,0 +1,21 @@
+#   - Resources/Textures/_FarHorizons/Mobs/Customization/ipc/shellguard.rsi/
+#   - Resources/Textures/_FarHorizons/Mobs/Customization/ipc/wardtakahashi.rsi/
+#   - Resources/Textures/_FarHorizons/Mobs/Customization/ipc/xion.rsi/
+#   - Resources/Textures/_FarHorizons/Mobs/Customization/ipc/zenghu.rsi/
```
Like wtf is this, why you dont PUT ALL THIS 200+ files in _FarHorizons Folders at first place?

CringeCursed commented on Resources/ServerInfo/Guidebook/Mobs/Species.xml:
while "you" on it wrap all our species in starlight comment

CringeCursed commented on Resources/Prototypes/Voice/speech_emotes.yml:
```diff
@@ -422,6 +422,12 @@
+  name: chat-emote-name-deathgasp
+  icon: Interface/Emotes/deathgasp.png
+  chatMessages: ["chat-emote-msg-deathgasp-ipc"]
+
```
to FH folder

CringeCursed commented on Resources/Prototypes/Entities/Mobs/Species/ipc.yml:
to FH folder

CringeCursed commented on Resources/Prototypes/Entities/Mobs/Player/ipc.yml:
to FH folder

CringeCursed commented on Resources/Prototypes/Entities/Mobs/Customization/Markings/ipc_wings.yml:
to FH folder

CringeCursed commented on Resources/Prototypes/Entities/Mobs/Customization/Markings/ipc.yml:
to FH folder

CringeCursed commented on Resources/Prototypes/Damage/containers.yml:
Comments

CringeCursed commented on Resources/Prototypes/Body/Organs/ipc.yml:
to FH folder

CringeCursed commented on Resources/Locale/en-US/silicons/alerts.ftl:
to FH folder

CringeCursed commented on Resources/Locale/en-US/medical/components/healing-component.ftl:
```diff
@@ -1,5 +1,6 @@
 ﻿medical-item-finished-using = You have finished healing with the {$item}.
 medical-item-cant-use = There is no damage you can heal with the {$item}.
+medical-item-not-activated = You need to activate {THE($item)} first!
```
atleast add comments, or better move to corresponding folder

CorruptOmega commented:
Still a good few things need to be moved from wizden/Starlight directories back to Far Horizons.

CorruptOmega commented on Resources/Locale/en-US/_Starlight/alerts/ipc.ftl:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Locale/en-US/_Starlight/battery/components/battery-drainer-component.ftl:
Should move this new line to _Starlight directory.

CorruptOmega commented on Resources/Locale/en-US/chat/emotes.ftl:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Locale/en-US/_Starlight/markings/ipc.ftl:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Locale/en-US/medical/components/healing-component.ftl:
Needs to be moved to _Starlight directory.

CorruptOmega commented on Resources/Locale/en-US/silicons/alerts.ftl:
Needs to be moved to _Starlight directory.

CorruptOmega commented on Resources/Locale/en-US/_Starlight/silicons/ipc.ftl:
Needs to be moved to _Starlight directory. Seem to be missing this stuff downstream.

CorruptOmega commented on Resources/Locale/en-US/species/species.ftl:
New line needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/_StarLight/Alerts/ipc.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/_StarLight/Body/Organs/ipc.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/_StarLight/Body/Parts/ipc.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/_StarLight/Body/Prototypes/ipc.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/Damage/containers.yml:
Need #Starlight comments.

CorruptOmega commented on Resources/Prototypes/Damage/modifier_sets.yml:
Need comment block start/end. 
#Starlight-Start
-damageModifierSet
etc
etc
#Starlight-End

CorruptOmega commented on Resources/Prototypes/Damage/types.yml:
Add comment block #Starlight-Start/#Starlight-End

CorruptOmega commented on Resources/Prototypes/_StarLight/Entities/Mobs/Customization/Markings/ipc.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/_StarLight/Entities/Mobs/Customization/Markings/ipc_wings.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/_StarLight/Entities/Mobs/Player/ipc.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/_StarLight/Entities/Mobs/Species/ipc.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/Guidebook/species.yml:
Need FarHorizons comments

CorruptOmega commented on Resources/Prototypes/_StarLight/Species/ipc.yml:
Needs to be moved to _FarHorizons directory.

CorruptOmega commented on Resources/Prototypes/Voice/speech_emotes.yml:
Needs to be moved to _FarHorizons directory, preferably with matching what we already had.

- type: emote
  id: DeathgaspIPC
  name: chat-emote-name-deathgasp
  icon: Interface/Emotes/deathgasp.png
  whitelist:
    components:
    - MobState
  chatMessages: ["chat-emote-msg-deathgaspipc"]

CorruptOmega commented on Resources/ServerInfo/Guidebook/_Starlight/Mobs/IPC.xml:
Needs to be moved to _FarHorizons directory. Also should be updated downstream.

CorruptOmega commented on Resources/ServerInfo/Guidebook/Mobs/Species.xml:
Make this a FH comment.

CorruptOmega commented on Resources/Textures/_Starlight/Mobs/Customization/ipc/antenna.rsi/ipc_antenna_antlers.png:
All textures need to be moved to _FarHorizons directory under matching files as what we got downstream.

ScarletLightweaverSS14 committed:
moved files and added comments

ScarletLightweaverSS14 committed:
Merge branch 'IPC' of https://github.com/ScarletLightweaverSS14/Starlight-Chimes into IPC

ScarletLightweaverSS14 commented:
done i think @CorruptOmega

CorruptOmega commented on Resources/Prototypes/Guidebook/species.yml:
```diff
@@ -10,6 +10,7 @@
   - Dwarf
   - Felionoid
   - Human
+  - IPC
```
```suggestion
  - IPC #Far Horizons
```

CorruptOmega commented on Resources/Prototypes/Guidebook/species.yml:
```diff
@@ -39,6 +40,11 @@
   name: species-name-human
   text: "/ServerInfo/Guidebook/Mobs/Human.xml"
 
+- type: guideEntry
```
```suggestion
- type: guideEntry #Far Horizons
```

CorruptOmega commented on Resources/Prototypes/Entities/Objects/Tools/cable_coils.yml:
```diff
@@ -45,6 +45,16 @@
     endSound:
       path: /Audio/_Starlight/Medical/Surgery/retractor2.ogg
   # 🌟Starlight-end🌟
+  - type: Healing # _STARLIGHT: Silicon repair - cable coil heals electrical/thermal damage on silicons (IPCs, borgs)
```
```suggestion
  - type: Healing #Far Horizons: Silicon repair - cable coil heals electrical/thermal damage on silicons (IPCs, borgs)
```

CorruptOmega commented on Resources/Prototypes/Damage/containers.yml:
```diff
@@ -27,10 +27,13 @@
   supportedGroups:
-    - Brute
+    - Brute # Blunt, Slash, Piercing damage
+    - Burn # Heat, Shock, Cold, Caustic damage
```
```suggestion
    - Burn #Starlight Heat, Shock, Cold, Caustic damage
```

CorruptOmega commented on Resources/Prototypes/Damage/containers.yml:
```diff
@@ -27,10 +27,13 @@
   supportedTypes:
     - Heat
     - Shock
+    - Caustic
```
```suggestion
    - Caustic #Starlight
```

CorruptOmega commented on Resources/Prototypes/Damage/containers.yml:
```diff
@@ -27,10 +27,13 @@
     - Heat
     - Shock
+    - Caustic
+    - Cold
```
```suggestion
    - Cold #Starlight
```

CorruptOmega commented on Resources/Prototypes/Damage/modifier_sets.yml:
```diff
@@ -10,6 +10,21 @@
+    Cold: 0.95      # 5% resistance - cold affects performance but doesn't damage as much
+    Shock: 2.0      # 100% extra damage (2x total) - electrical damage is dangerous to circuits but not instant death
+    Caustic: 1.1    # 10% weakness - acid corrodes metal and circuits
+
```
```suggestion
#Starlight-End

```

CorruptOmega commented on Resources/Prototypes/Damage/types.yml:
```diff
@@ -71,6 +71,7 @@
   armorCoefficientPrice: 2.5
   armorFlatPrice: 20
 
+# Ion damage - specifically for silicons (IPCs, borgs)
```
```suggestion
# Starlight - Ion damage - specifically for silicons (IPCs, borgs)
```

CorruptOmega commented on Resources/Prototypes/Damage/modifier_sets.yml:
```diff
@@ -10,6 +10,21 @@
     Blunt: 5
     Heat: 5
 
+# _STARLIGHT: IPC damage modifiers - Balanced for mechanical beings
```
```suggestion
# _STARLIGHT-Start: IPC damage modifiers - Balanced for mechanical beings
```

CorruptOmega commented:
Going to be some other stuff Starlight will likely need commenting on. But good portion of FH stuff looking good.

Also going to need to redirect the sprite paths to new folder directory to appease the now angry YAML gods.

ScarletLightweaverSS14 committed:
patch

ScarletLightweaverSS14 commented:
@CorruptOmega  i hope i didnt miss anythin

ScarletLightweaverSS14 committed:
Update Resources/Prototypes/Damage/modifier_sets.yml

Co-authored-by: CorruptOmega <207436787+CorruptOmega@users.noreply.github.com>

ScarletLightweaverSS14 committed:
Update Resources/Prototypes/Damage/modifier_sets.yml

Co-authored-by: CorruptOmega <207436787+CorruptOmega@users.noreply.github.com>

ScarletLightweaverSS14 committed:
Update Resources/Prototypes/Damage/containers.yml

Co-authored-by: CorruptOmega <207436787+CorruptOmega@users.noreply.github.com>

ScarletLightweaverSS14 committed:
Update Resources/Prototypes/Damage/containers.yml

Co-authored-by: CorruptOmega <207436787+CorruptOmega@users.noreply.github.com>

ScarletLightweaverSS14 committed:
Update Resources/Prototypes/Damage/containers.yml

Co-authored-by: CorruptOmega <207436787+CorruptOmega@users.noreply.github.com>

ScarletLightweaverSS14 committed:
Update Resources/Prototypes/Entities/Objects/Tools/cable_coils.yml

Co-authored-by: CorruptOmega <207436787+CorruptOmega@users.noreply.github.com>

ScarletLightweaverSS14 committed:
Update Resources/Prototypes/Guidebook/species.yml

Co-authored-by: CorruptOmega <207436787+CorruptOmega@users.noreply.github.com>

ScarletLightweaverSS14 committed:
Update Resources/Prototypes/Guidebook/species.yml

Co-authored-by: CorruptOmega <207436787+CorruptOmega@users.noreply.github.com>

ScarletLightweaverSS14 committed:
Patch

ScarletLightweaverSS14 committed:
Merge branch 'starlight-dev' into IPC

ScarletLightweaverSS14 committed:
patch

ScarletLightweaverSS14 committed:
patch

ScarletLightweaverSS14 committed:
Patch

ScarletLightweaverSS14 committed:
Patch

beck-thompson commented on Content.Shared/_Starlight/DoAfterSpeedModifier/DoAfterSpeedModifierSystem.cs:
```diff
@@ -0,0 +1,44 @@
+
+        // Don't apply speed buff to self-healing (welding/cable repairs)
+        // This prevents IPCs from healing themselves faster
+        if (args.Args.Event is WelderHealingDoAfterEvent)
```
Hard coding it like this isn't very good, add it to some kind of dictionary blacklist or have an event thats called or something that IPCs can listen to and then can cancel if its welding. This will allow it to be a lot more flexible

beck-thompson commented on Content.Shared/_Starlight/DoAfterSpeedModifier/DoAfterSpeedModifierSystem.cs:
```diff
@@ -0,0 +1,44 @@
+/// Raised before a DoAfter is started, allowing systems to modify the args.
+/// </summary>
+[ByRefEvent]
+public record struct DoAfterStartAttemptEvent(DoAfterArgs Args);
```
This should be renamed, its not an attempt event! Attempt events can be canceled and shouldn't modify the args itself

CrazyPhantom779 commented:
Unfortunately it was decided that files being ported from downstream should go into _Starlight folders.

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingComponent.cs:
```diff
@@ -0,0 +1,32 @@
+    /// How much damage to heal per use
+    /// </summary>
+    [DataField]
+    public float HealAmount = 10f;
```
Should be one of those damage dictionaries, that way you don't need to hard code blunt into the system itself!

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingComponent.cs:
```diff
@@ -0,0 +1,32 @@
+    /// How long the welding takes
+    /// </summary>
+    [DataField]
+    public float Delay = 1.3f;
```
This should be a `TimeSpan`

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingSystem.cs:
```diff
@@ -0,0 +1,135 @@
+            return;
+
+        // Check if entity uses Silicon damage container
+        if (damageable.DamageContainerID != "Silicon")
```
What is this check for and are you sure its needed? If checks like this are necessary, it should be in the component as a list / dictionary or something.

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingSystem.cs:
```diff
@@ -0,0 +1,135 @@
+            return;
+
+        // Only run on server
+        if (_net.IsClient)
```
Does this misspredict? My gut feeling says it should work fine

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingSystem.cs:
```diff
@@ -0,0 +1,135 @@
+        var welder = args.Used.Value;   // The welder tool
+
+        // Get welder component and solution
+        if (!TryComp<WelderComponent>(welder, out var welderComp))
```
These checks should be in their own helper function, repeating them for before checking if you can start the do after and before leads to issues. E.g that ` if (damageable.DamageContainerID != "Silicon"` check isn't done on after the doafter is completed even though it probably should.

This PR has a good way of doing it https://github.com/space-wizards/space-station-14/pull/38505/ (You dont need the extra return values though, it can just be true false this is a lot simpler of a case)

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingSystem.cs:
```diff
@@ -0,0 +1,135 @@
+        _damageable.TryChangeDamage(target, damage, origin: args.User);
+
+        // Play sound and show popup
+        _audio.PlayPvs(new SoundPathSpecifier("/Audio/Items/welder.ogg"), target);
```
I dont like the audio being hard coded like this but because the welder should really be the one playing the noise there aren't any easy solutions

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingSystem.cs:
```diff
@@ -0,0 +1,135 @@
+
+        // Play sound and show popup
+        _audio.PlayPvs(new SoundPathSpecifier("/Audio/Items/welder.ogg"), target);
+        _popup.PopupEntity("Chassis repaired!", target, args.User);
```
This popup needs to be localized

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingSystem.cs:
```diff
@@ -0,0 +1,135 @@
+        _popup.PopupEntity("Chassis repaired!", target, args.User);
+
+        // Check if there's still damage to repair
+        if (TryComp<DamageableComponent>(target, out var damageable) && damageable.TotalDamage > 0)
```
This is going to be a little tricky, you are only healing blunt damge so this will break if you have damage other than blunt.

beck-thompson commented on Content.Shared/_Starlight/Silicons/WelderHealingSystem.cs:
```diff
@@ -0,0 +1,135 @@
+
+        // Heal damage - only Blunt type
+        var damage = new DamageSpecifier();
+        damage.DamageDict.Add("Blunt", FixedPoint2.New(-component.HealAmount));
```
See my comment on the component!

beck-thompson commented on Content.Shared/_Starlight/DoAfterSpeedModifier/DoAfterSpeedModifierSystem.cs:
```diff
@@ -0,0 +1,44 @@
+
+    private void OnDoAfterStartAttempt(EntityUid uid, DoAfterSpeedModifierComponent component, ref DoAfterStartAttemptEvent args)
+    {
+        if (component.SpeedModifier <= 0 || component.SpeedModifier == 1.0f)
```
Doing floating point comparisons like this is a little tricky and not really worth it! Its probably better to skip the equality check

beck-thompson commented on Content.Shared/_Starlight/Silicons/HumanoidEMP/HumanoidEMPComponent.cs:
```diff
@@ -0,0 +1,73 @@
+    /// How long to stun the entity when hit by EMP (in seconds).
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public float StunTime = 5f;
```
Use a `TimeSpan`

beck-thompson commented on Content.Shared/_Starlight/Silicons/HumanoidEMP/HumanoidEMPComponent.cs:
```diff
@@ -0,0 +1,73 @@
+    /// How long to knock down the entity (in seconds).
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public float KnockdownTime = 0f;
```
Same here, use a `TimeSpan`

beck-thompson commented on Content.Shared/_Starlight/Silicons/HumanoidEMP/HumanoidEMPComponent.cs:
```diff
@@ -0,0 +1,73 @@
+    /// How long to slow down the entity (in seconds).
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public float SlowdownTime = 0f;
```
time span!

beck-thompson commented on Content.Shared/_Starlight/Silicons/HumanoidEMP/HumanoidEMPSystem.cs:
```diff
@@ -0,0 +1,86 @@
+        }
+
+        // Slowdown would need to be implemented through MovementModStatusSystem
+        // if (comp.SlowdownTime > 0)
```
Remove code thats commented out like this

beck-thompson commented on Content.Shared/_Starlight/Silicons/HumanoidEMP/HumanoidEMPComponent.cs:
```diff
@@ -0,0 +1,73 @@
+/// Allows for stuns, knockdowns, damage, and other effects.
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class HumanoidEMPComponent : Component
```
The best way to do this would be to split this up into like 3-4 different components one for each property. It would be a little annoying though because each one is only used for this one species 😆

Talk to the maintainers about this it might be easier to maintain if its just one big file component "wrong"

beck-thompson commented on Content.Shared/_Starlight/Silicons/HumanoidEMP/HumanoidEMPSystem.cs:
```diff
@@ -0,0 +1,86 @@
+        // }
+
+        // Apply damage
+        if (comp.Damage != null && TryComp<DamageableComponent>(ent, out var damageable))
```
You don't need these trycomps, you can just let the system resolve it!

beck-thompson commented on Content.Server/_Starlight/Temperature/KillOnOverheatSystem.cs:
```diff
@@ -0,0 +1,82 @@
+                    var msg = Loc.GetString(comp.OverheatPopup, ("name", Identity.Name(uid, EntityManager)));
+                    _popup.PopupEntity(msg, uid, PopupType.LargeCaution);
+                    
+                    // Knock out the entity (not kill, so they can recover)
```
probably better to rename the system and component if it doesn't actually kill 😆

beck-thompson commented on Content.Server/_Starlight/Temperature/KillOnOverheatSystem.cs:
```diff
@@ -0,0 +1,82 @@
+                    _popup.PopupEntity(msg, uid, PopupType.LargeCaution);
+                    
+                    // Knock out the entity (not kill, so they can recover)
+                    _mob.ChangeMobState(uid, MobState.Critical, mob);
```
it would be better to put this in the component so it could be customized (E.g you could set it to dead!)

beck-thompson commented on Content.Server/_Starlight/Temperature/KillOnOverheatSystem.cs:
```diff
@@ -0,0 +1,82 @@
+            else
+            {
+                // Temperature has dropped below threshold, reset overheat flag
+                if (comp.HasOverheated)
```
Doing switching like this is difficult and can lead to bugs easily, it would probably be better to just do this based off if they are in a critical state or not.

beck-thompson commented on Content.Server/_FarHorizons/Silicons/IPC/IPCSystem.Revive.cs:
```diff
@@ -0,0 +1,222 @@
+
+        // Deal 50 shock damage to the IPC (damages them instead of helping)
+        var damage = new DamageSpecifier();
+        damage.DamageDict.Add("Shock", 50);
```
Add this to the component as a damage dictionary

beck-thompson commented on Content.Server/_FarHorizons/Silicons/IPC/IPCSystem.Revive.cs:
```diff
@@ -0,0 +1,222 @@
+        _damageable.TryChangeDamage(ent.Owner, damage);
+        
+        // Electrocute the user (125 damage, 5 second stun)
+        _electrocution.TryDoElectrocution(args.EntityUsingDefib, null, 125, TimeSpan.FromSeconds(5), true);
```
Put those values in the component as well

beck-thompson commented on Content.Server/_FarHorizons/Silicons/IPC/IPCSystem.Revive.cs:
```diff
@@ -0,0 +1,222 @@
+    {
+        if (!ev.CanInteract || !ev.CanAccess || !ev.CanComplexInteract ||
+            // IPCLockComponent not implemented yet, skip lock check
+            // !TryComp<IPCLockComponent>(ev.Target, out var lockComp) ||
```
Remove unused comments

beck-thompson reviewed (commented):
a quick review, I didn't look at everything

ScarletLightweaverSS14 committed:
patch

princess-gurchi commented:
> Unfortunately it was decided that files being ported from downstream should go into _Starlight folders.

So, a feature developed by 2 people that were driven away from SL development because of poor treatment are going to have their credit erased when ported to SL. Not to mention creating further issues with maintaining downstream for no benefit to upstream. All the while there are other features from other downstreams that get to keep their credit. Am I getting it right?

ScarletLightweaverSS14 commented:
> > Unfortunately it was decided that files being ported from downstream should go into _Starlight folders.
> 
> So, a feature developed by 2 people that were driven away from SL development because of poor treatment are going to have their credit erased when ported to SL. Not to mention creating further issues with maintaining downstream for no benefit to upstream. All the while there are other features from other downstreams that get to keep their credit. Am I getting it right?


i feel you. if this is the case, im closing the PR

its disgusting to do this. and im not gonna mark it ready til its actually good to go. erasing credits is horrible

ScarletLightweaverSS14 commented:
As of now its drafted.

CringeCursed commented:
> Unfortunately it was decided that files being ported from downstream should go into _Starlight folders.

Wait what? Who decided that? Why?

ItsAbby11 commented:
> Unfortunately it was decided that files being ported from downstream should go into _Starlight folders.

I need an official confirmation from the Maintainer or head developer of Starlight, but if this happens to be true, i can't say how much this piss me off.

1- Two developers from Far Horizons made this stuff, the code was written by them, and just moving that content to the SL Folder like you guys made it in the first place is a real dick move.

2- There is 0 Regard for the downstream forks like us, considering this will cause a massive technical hellscape everytime we merge SL content to our server.

If this again happens to be true, we will cut all ties with Starlight, we refuse to cooperate with people who use this type of tactics.

ScarletLightweaverSS14 commented:
> > Unfortunately it was decided that files being ported from downstream should go into _Starlight folders.
> 
> I need an official confirmation from the Maintainer or head developer of Starlight, but if this happens to be true, i can't say how much this piss me off.
> 
> 1- Two developers from Far Horizons made this stuff, the code was written by them, and just moving that content to the SL Folder like you guys made it in the first place is a real dick move.
> 
> 2- There is 0 Regard for the downstream forks like us, considering this will cause a massive technical hellscape everytime we merge SL content to our server.
> 
> If this again happens to be true, we will cut all ties with Starlight, we refuse to cooperate with people who use this type of tactics.




If maintainers do not go back on what they said im closing this PR.

princess-gurchi commented:
@Happyrobot33 
So it came to my attention that you are confused about my response. Which, weirdly enough, you refuse to communicate on a public forum where my response is posted. So here's me being as clear as possible.

Let's drop the concerns for maintaing downstreams. It's an annoyance but not the end of the world, we can handle it. 

The only real concern I have is taking the credit. 

You claim what you doing is acceptable because there's still records in the way of git log and patchnotes. It's very unfortunate that I have to argue from the position of having my name there. I would have had much higher ground to stand on if it wasn't. However, you just have to believe me when I say my concern isn't about my name being forgotten. My concern is for the server Far Horizons and it's community. As well as every single other downstream of Starlight.

It's established in ss14 community how features are ported. Sure there are a lot of licenses and a lot of hurt feelings, but take a look at this:

https://github.com/Goob-Station/Goob-Station/tree/master/Resources/Prototypes/_StarLight

Interesting, is that a `_StarLight` folder on goob station? I wonder where it came from, I probably should check git log or dig through patch notes to find out.

What about this

https://github.com/Monolith-Station/Monolith/tree/main/Resources/Prototypes/_StarLight/SoundCollections

A single file in the entire `_StarLight` directory, 8 lines of yaml. And yet it's clear and immediately visible. 

That's how it is wherever you go, any space station 14 fork. 

**That's how everyone expects it to be**.

Now riddle me this. It's very unlikely that any Starlight downstream is going to be bigger than Starlight. Starlight is the biggest server there is. So it's an objective fact that any feature is more likely to be discovered on Starlight than any of its downstreams.

How do you think it's going to be ported? Do you think everyone will keep git log? Just like it's kept in this PR with my entire code under porter's name? Everyone's patchnotes will reference it?

Happyrobot33 commented:
Todo list before this can merge, needs first commit rebased to have author edited properly

Happyrobot33 commented:
Todo list before this can merge, needs first commit rebased to have author edited properly

ScarletLightweaverSS14 commented:
> Todo list before this can merge, needs first commit rebased to have author edited properly

are you gonna do that or do i need to

ScarletLightweaverSS14 commented:
also i still havnt done the alt interact for IPC charging from an APC

Happyrobot33 commented:
> > Todo list before this can merge, needs first commit rebased to have author edited properly
> 
> are you gonna do that or do i need to

Easier if you do it since it involves a history rewrite. While your at it rename the initial commit to involve the fh pr number

ScarletLightweaverSS14 commented:
> 135

how do i do it

ScarletLightweaverSS14 commented:
@Happyrobot33

ScarletLightweaverSS14 committed:
Fix IPC component duplication and add substation charging (FH#135)

ScarletLightweaverSS14 commented:
Okay so i will NOT be letting this project go. **HOWVER** i want to make this clear.
Until i get word from @ItsAbby11  @KillerTheFlareon  or other maintainers from FH or SL i will NOT be working on it.


we need to know what to do first

ScarletLightweaverSS14 commented:
@Happyrobot33  update?
@KillerTheFlareon 
@princess-gurchi

Rinary1 commented:
Marked as stale

ScarletLightweaverSS14 commented:
Due to changes neither I or the server I'm porting IPC from respect recent changes that make it extremely difficult for FH to get updates and nearly erasing all credit from FH files I'm closing this pr

neomoth commented:
Personally I completely agree that moving things to an SL directory is entirely bullshit.
Not only does it work as a form of attribution to keep it in its own directory but it also makes future code changes from FH much easier to pull upstream.
There is literally zero reason this ever should have been an argument, moving things into the SL directory that do not need to be in the SL directory is fucking stupid and only makes things harder to maintain when they change. It's the same reason we don't just put everything in upstream directories and files and have comments whenever we do edit, its to avoid conflicts when pulling from upstream. Same should apply when pulling from downstream. ZERO fucking reason for any of this to not be in a FH directory.

ScarletLightweaverSS14 commented:
Gamers
 It's time. 

I'm moving the files into FH 
@Rinary1 please unstale 
@walksanatora  can I poke you to review current IPC. point out godos and bados

ScarletLightweaverSS14 commented:
Well what a bummer.
I thought we could have fun with playing IPC, no it won't go that way. Sadly after seeing a screenshot.

I'm disappointed

StarlightHost commented:
The folder where a feature lives is not about attribution, it is about who maintains it.

Code that lives in other folders like _AL _CD _TP must also be marked with comments like //starlight, because maintenance responsibility remains with those projects.

If something is planned to be continuously updated further, basically upstreamed, and fixes are done by the fork developers, it goes into the fork folder. If we take control and continue development ourselves, it goes into our folder.

The fact that other forks follow different rules is their choice. We do this to minimize conflicts.

If this does not suit you, create a ticket in Discord.

One day, we will most likely get rid of the _Starlight folder entirely, take full maintenance responsibility for all the code, and start porting WizDen features into _Wizden folders instead.

ScarletLightweaverSS14 committed:
Merge upstream/starlight-dev and update IPC credits

- Updated all IPC file headers with creator credits: Killer Tamashi and Princess Gurchi
- Resolved merge conflicts from upstream/starlight-dev
- Removed redundant code from IPC battery and temperature systems
- Fixed BatteryDrainerSystem imports
- Accepted upstream changes for deleted files (lobby-gui.ftl, gateway.yml)

ScarletLightweaverSS14 committed:
Update to .NET 10

- Updated RobustToolbox to latest master (fe1648d29) for .NET 10 support
- Commented out missing test projects from solution file
- Fixed EntityFrameworkCore 10.0.0 compatibility

ScarletLightweaverSS14 commented:
test running...

ScarletLightweaverSS14 commented:
yay ok

ScarletLightweaverSS14 committed:
Fix API compatibility issues after RobustToolbox .NET 10 update

- Replace PredictedBatterySystem with BatterySystem (PredictedBatterySystem was removed)
- Add using Content.Shared.Gibbing for BeingGibbedEvent
- Remove obsolete .Playing field check in IPCSystem.Revive.cs
- Fix AudioSystem.PlayPvs API change in SharedIPCSystem.SparkEffects.cs

All changes are in IPC code only, no RobustToolbox modifications.

ScarletLightweaverSS14 committed:
Fix duplicate subscription error for CyborgSparkEffectComponent

Remove duplicate DamageChangedEvent subscription from ArmorSparkEffectSystem.
The event is now only handled by SharedIPCSystem to avoid duplicate
subscription errors during system initialization.

ScarletLightweaverSS14 committed:
Merge branch 'starlight-dev' into IPC

ScarletLightweaverSS14 committed:
wawaa

ScarletLightweaverSS14 committed:
Merge branch 'IPC' of https://github.com/ScarletLightweaverSS14/Starlight-Chimes into IPC

ScarletLightweaverSS14 committed:
Merge branch 'starlight-dev' into IPC

ScarletLightweaverSS14 committed:
wawa

ScarletLightweaverSS14 committed:
Merge branch 'IPC' of https://github.com/ScarletLightweaverSS14/Starlight-Chimes into IPC

ScarletLightweaverSS14 commented:
hi sorry i havnt been active, ive been demoltivated

Conflee reviewed (approved):
sprites approved, I've seen them in use elsewhere thank god so I don't have to comb over every png manually.

ScarletLightweaverSS14 commented:
real.

okay also im gonna update and fix the PR somwhat next week. ive been bummed out quite a bit from big projects

ScarletLightweaverSS14 committed:
Merge branch 'starlight-dev' into IPC

ScarletLightweaverSS14 committed:
Merge branch 'starlight-dev' into IPC

Rinary1 commented:
Marked as stale
